src/Microsoft.Health.Fhir.Core.UnitTests/Features/Operations/Import/ImportOrchestratorJobTests.cs

```diff
@@ -68,16 +86,561 @@ public async Task GivenAnOrchestratorJobAndWrongEtag_WhenOrchestratorJobStart_Th
+            IMediator mediator = Substitute.For<IMediator>();
+            ImportOrchestratorJobInputData importOrchestratorInputData = new ImportOrchestratorJobInputData();
+            List<(long begin, long end)> surrogatedIdRanges = new List<(long begin, long end)>();
+
+            importOrchestratorInputData.CreateTime = Clock.UtcNow;
+            importOrchestratorInputData.BaseUri = new Uri("http://dummy");
+            var inputs = new List<InputResource>();
+            inputs.Add(new InputResource() { Type = "Resource", Url = new Uri($"http://dummy") });
+
+            importOrchestratorInputData.Input = inputs;
+            importOrchestratorInputData.InputFormat = "ndjson";
+            importOrchestratorInputData.InputSource = new Uri("http://dummy");
+            importOrchestratorInputData.RequestUri = new Uri("http://dummy");
+
+            integrationDataStoreClient.GetPropertiesAsync(Arg.Any<Uri>(), Arg.Any<CancellationToken>())
+                .Returns(callInfo =>
+                {
+                    Dictionary<string, object> properties = new Dictionary<string, object>();
+                    properties[IntegrationDataStoreClientConstants.BlobPropertyETag] = "test";
+                    properties[IntegrationDataStoreClientConstants.BlobPropertyLength] = 1000L;
+                    return properties;
+                });
+
+            fhirDataBulkImportOperation.PreprocessAsync(Arg.Any<CancellationToken>())
+                .Returns(_ =>
+                {
+                    throw new RetriableJobException("test");
+                });
+
+            TestQueueClient testQueueClient = new TestQueueClient();
+            JobInfo orchestratorJobInfo = (await testQueueClient.EnqueueAsync(0, new string[] { JsonConvert.SerializeObject(importOrchestratorInputData) }, 1, false, false, CancellationToken.None)).First();
+
+            ImportOrchestratorJob orchestratorJob = new ImportOrchestratorJob(
+                mediator,
+                importOrchestratorInputData,
+                new ImportOrchestratorJobResult(),
+                contextAccessor,
+                fhirDataBulkImportOperation,
+                integrationDataStoreClient,
+                testQueueClient,
+                orchestratorJobInfo,
+                new Configs.ImportTaskConfiguration() { MaxRunningProcessingJobCount = 1 },
+                loggerFactory);
+            orchestratorJob.PollingFrequencyInSeconds = 0;
+
+            await Assert.ThrowsAnyAsync<RetriableJobException>(() => orchestratorJob.ExecuteAsync(new Progress<string>(), CancellationToken.None));
+
+            _ = mediator.DidNotReceive().Publish(
+                Arg.Any<ImportJobMetricsNotification>(),
+                Arg.Any<CancellationToken>());
+        }
+
+        [Fact]
+        public async Task GivenAnOrchestratorJob_WhenLastSubJobFailed_ThenImportProcessingExceptionShouldBeThrowAndWaitForOtherSubJobsCompleted()
```

Comment:
>WaitForOtherSubJobsCompleted [](http://example.com/codeflow?start=115&length=28)

Other jobs should be cancelled and completed.

Reply:
Sorry, I will modify it.
ThenImportProcessingExceptionShouldBeThrowAndWaitForOtherSubJobsCancelledAndCompleted